glitter/pages/admin.py

```diff
@@ -71,5 +77,64 @@ def wrapper(*args, **kwargs):
         urlpatterns = [
             url(r'^$', wrap(self.grid_view)),
             url(r'^tree/$', wrap(self.changelist_view), name='%s_%s_tree' % info),
+            url(r'^(\d+)/duplicate/$', wrap(self.duplicate_page), name='%s_%s_duplicate' % info),
         ] + urlpatterns
         return urlpatterns
+
+    def duplicate_page(self, request, obj_id):
+        obj = get_object_or_404(Page, id=obj_id)
+
```

Comment:
Need to check if they've got the add permission here, otherwise anyone could create a page.